lib/customer-deposit-wallet/src/Cardano/Wallet/Deposit/Pure.hs

```diff
@@ -230,16 +237,32 @@ getWalletTip :: WalletState -> Read.ChainPoint
 getWalletTip = walletTip
 
 rollForwardMany
-    :: NonEmpty (Read.EraValue Read.Block) -> WalletState -> WalletState
-rollForwardMany blocks w = foldl' (flip rollForwardOne) w blocks
+    :: ResolveSlot
+    -> NonEmpty (Read.EraValue Read.Block)
+    -> WalletState
+    -> WalletState
+rollForwardMany resolveSlot blocks w =
+    foldl' (flip $ rollForwardOne resolveSlot) w blocks
 
 rollForwardOne
-    :: Read.EraValue Read.Block -> WalletState -> WalletState
-rollForwardOne (Read.EraValue block) w =
+    :: ResolveSlot
+    -> Read.EraValue Read.Block
+    -> WalletState
+    -> WalletState
+rollForwardOne resolveSlot eblock@(Read.EraValue block) w =
     w
         { walletTip = Read.getChainPoint block
         , utxoHistory = rollForwardUTxO isOurs block (utxoHistory w)
         , submissions = Delta.apply (Sbm.rollForward block) (submissions w)
+        , txHistory =
+            TxHistory.rollForward
+                ( valueTransferFromResolvedTx . resolveInputs
+                    (getUTxO $ utxoHistory w)
```

Comment:
I think that this is incorrect when the block contains multiple transactions that depend on each other — `getUTxO $ utxoHistory w` will contain the correct UTxO set at the beginning of the block, but not in-between transactions. 🤓 

Could you augment `Cardano.Wallet.Deposit.Pure.Balance` to compute the data that you need? 

I recommend to modify / augment the `applyBlock` function from to compute the value transfers for each individual transaction. Then, you can pass the computed data as `Map TxId (Map CompactAddr ValueTransfer)` to `TxHistory.rollForward`.

We need to be careful about the correctness of this data, hence I want to isolate its computation in `Cardano.Wallet.Deposit.Pure.Balance` and property test it there.